client-mr/src/main/java/org/apache/hadoop/mapreduce/task/reduce/RssBypassWriter.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Tencent is pleased to support the open source community by making
+ * Firestorm-Spark remote shuffle server available.
+ *
+ * Copyright (C) 2021 THL A29 Limited, a Tencent company.  All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at
+ *
+ * https://opensource.org/licenses/Apache-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package org.apache.hadoop.mapreduce.task.reduce;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.lang.reflect.Field;
+import java.nio.ByteBuffer;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.hadoop.io.IOUtils;
+import org.apache.hadoop.mapreduce.TaskAttemptID;
+
+import com.tencent.rss.common.exception.RssException;
+
+// In MR shuffle, MapOutput encapsulates the logic to fetch map task's output data via http.
+// So, in RSS, we should bypass this logic, and directly write data to MapOutput.
+public class RssBypassWriter {
+  private static final Log LOG = LogFactory.getLog(RssBypassWriter.class);
+
+  public static void write(MapOutput mapOutput, ByteBuffer buffer,
+                           TaskAttemptID reduceId, long rssBlockId) {
```

Comment:
I think we should remove blockId, and Compressed data also remove blockId.

Reply:
you mean "write(MapOutput mapOutput, ByteBuffer buffer)"?

Reply:
Yes

Reply:
Done